package/base-files/files/bin/board_detect

```diff
@@ -1,14 +1,20 @@
 #!/bin/sh
 
-CFG=$1
+CFG="${1:-/etc/board.json}"
 
-[ -n "$CFG" ] || CFG=/etc/board.json
-
-[ -d "/etc/board.d/" -a ! -s "$CFG" ] && {
-	for a in $(ls /etc/board.d/*); do
-		[ -s $a ] || continue;
-		$(. $a)
+if [ -d '/etc/board.d/' ] && \
+   [ ! -s "${CFG}" ]; then
+	for _script in '/etc/board.d/'*; do
+		if [ ! -s "${_script}" ]; then
+			continue
+		fi
+		# shellcheck source=/dev/null disable=SC2091  # Can't use eval
+		$(. "${_script}")
```

Comment:
If we touch this code anyway, can we remove the leading dollar? The intention seems to be spawning a subshell to avoid polluting the current environment, output is not used.

Reply:
So here is where it becomes difficult.

For whatever reason, the initial commit 562d9f958290acd4323b24fbecfec43514b392e9 (don't think John as a github handle?) introduced the dollar sign. Now, saying 'it is not used' is not true, because this dollar sign means, that any output from any script will get executed. Why? I have no idea. To me, it doesn't make sense (in the current context). I can imagine that initially, the idea was `board.d/*` generate code to be executed, but why, one would want to do that? I don't know.

Personally, I'd much rather put in the effort, on just executing this script, as you say (in a subshell, to not mess with the environment) without the leading dot as well (e.g. don't source it, just execute it. This would (almost) effectively revert 85b1f4d8ca2d501664acce471117dc612bb69d82 which was pushing more into the current direction, rather then the more sensible one (add execute bits, only run executable scripts, treat them as actual shell scripts. But why @adschm did this? According to the commit message, because we are sourcing files here. So why are we sourcing files? I don't know.

I suppose if we can get answers to these questions, we can go on and clean this bit up as well. But again, even removing the dollar, we need to know _why_ are we sourcing this and executing the output, was/is there any reason for doing so? Or was it just a mistake?